pkg/analysis_server/lib/src/services/correction/fix_internal.dart

```diff
@@ -1875,6 +1884,25 @@ class FixProcessor {
     }
   }
 
+  void _addFix_removeTypeName() {
+    final FunctionTypedFormalParameter functionTyped =
+        node is FunctionTypedFormalParameter
+            ? node
+            : node.getAncestor((node) => node is FunctionTypedFormalParameter);
+    if (functionTyped != null) {
+      _addReplaceEdit(rf.rangeNode(functionTyped),
+          utils.getNodeText(functionTyped.identifier));
+      _addFix(DartFixKind.REMOVE_TYPE, []);
+      return;
+    }
```

Comment:
Yes, but that is not going to happen, because the avoid using dynamic only can be linted in a parameter of a function expression, it can not happen in the parameter of a function-typed formal parameter, so it is the right solution, the only case that this would fail if it would be necessary a fix to remove a typeName inside a function-typed formal parameter, and for now there is not lint that requires that